### DISCUSSION TITLE
The purpose of the default export component

### DISCUSSION THREAD

**Discussion by cgatian** [storybookjs/storybook#16954] [Help] [answered]
The docs state to set the `component` property of the default export, that makes sense! However, I have some stories that are actually wrapping the component and sometimes exposing a completely new API.

For example:

```ts
export default {
  component: Button,  // <= Questioning if this is actually the right component to use
};

interface NeatButtonStoryProps {
   title: string;
   disabled: boolean;
}

const NeatButtonStory: React.VFC<NeatButtonStoryProps> = ({ title, disabled }) => {
       // other setup code
      return (
               <Wrapper title={title}>
                         <Button disabled={disabled} />
               </Wrapper>
      )
}

export const Default: ComponentStoryObj<typeof NeatButtonStory> = {
  args: {
    title: 'Good job!'
  },
};
```

The `Button` is no longer the main actor within the story as other components/providers are augmenting the nature of the story. The validity of the story's composition aside (i.e., why isn't `NeatButtonStory` its own component). I think it makes sense to change the  `component` property in the default export to `NeatButtonStory`.

The Storybook Documentation references the creation of a `Template` component, which from what I've seen typically doesn't get too extravagant, and I _think_ should only be used when you need to map the `args` differently.

Only docs I could find on the prop usage:
> The component field is required and used by addons for automatic prop table generation and display of other component metadata. The title field is optional and should be unique (i.e., not re-used across files).
[CSF-default-export](https://storybook.js.org/docs/react/api/csf#default-export)

Anyways I feel like I'm stuck in between what's the best guidance here. I think the root cause is there should actually be a component that does all this wrapping, and THATS what the story should be interfacing with. That aside, is there any problem swapping `Button` as the component type for `NeatButtonStory`?

Note: I also want the Storybook Controls to use the `NeatButtonStoryProps`, not the actual props from the `Button` component.

**Comment by shilman** [comment, marked as answer]
For now, the primary information that's extracted from `default.component` is the `ArgTypes` which generates the controls/ArgsTable. However, I don't think you can just put a story into that field (haven't tried it, but I'd be surprised). So if your story has very different args from the underlying component, I'd just manually specify the `argTypes` per the documentation. https://storybook.js.org/docs/react/essentials/controls#fully-custom-args

**Reply by cgatian** [reply]
Sorry when I say Story I'm referring to a Template component. That was my fault.

If I understand correctly, with my example above, since NeatButtonStory (a component) has a different set of argTypes than Button, I could specify argTypes as you suggested, or I could use NeatButtonStory as the component in the default export. Both would be acceptable, correct?

**Reply by shilman** [reply]
>  or I could use NeatButtonStory as the component in the default export. 

I don't think so per my comment above: "I don't think you can just put a story into that field (haven't tried it, but I'd be surprised)"

**Reply by cgatian** [reply]
I could be completely mistaken, but isn't NeatButtonStory (sorry this is probably poor naming on my part) just a wrapper component to the underlying button? 

Thanks for taking the time. The CSF V3 syntax is great.

**Reply by cgatian** [reply]
I suppose the best guidance is to use the component you're writing the story as the default `component`. If you end up wrapping that or providing additional args, continue to use the original component.

**Reply by shilman** [reply]
Yeah that sounds right to me. Sorry this all isn't more clear--I'm hoping we can make it better next year.
